src/matcha_ml/state/matcha_state.py

```diff
@@ -195,41 +194,72 @@ def _parse_terraform_output_resource_name(
             Returns:
                 Tuple[str, str, str]: the resource output for matcha.state.
             """
-            resource_type: Optional[str] = None
+            resource_type = None
 
             for key in RESOURCE_NAMES:
                 if key in output_name:
-                    resource_type = key
+                    resource_type = MatchaResource(key)
                     break
 
             if resource_type is None:
                 raise MatchaInputError(
                     f"A valid resource type for the output '{output_name}' does not exist."
                 )
 
-            flavor_and_resource_name = output_name[len(resource_type) + 1 :]
+            flavor_and_resource_name = output_name[len(resource_type.name) + 1 :]
 
             flavor, resource_name = flavor_and_resource_name.split("_", maxsplit=1)
             resource_name = resource_name.replace("_", "-")
-            resource_type = resource_type.replace("_", "-")
+            resource_type = resource_type.name.replace("_", "-")  # type: ignore
 
             return resource_type, flavor, resource_name
 
-        state_outputs: Dict[str, Dict[str, str]] = defaultdict(dict)
+        matcha_components: List[MatchaStateComponent] = []
```

Comment:
N.b. can we use the MathcaState object here and then the associated find_component method later on?